cli/remotebazel/remotebazel.go

```diff
@@ -779,28 +783,35 @@ func Run(ctx context.Context, opts RunOpts, repoConfig *RepoConfig) (int, error)
 	log.Debugf("Invocation ID: %s", iid)
 
 	// If the remote bazel process is canceled or killed, cancel the remote run
-	sigChan := make(chan os.Signal, 1)
+	isInvocationRunning := true
 	go func() {
-		<-sigChan
-		_, err = bbClient.CancelExecutions(ctx, &inpb.CancelExecutionsRequest{
+		<-ctx.Done()
+
+		if !isInvocationRunning {
+			return
+		}
+
+		// Use a non-cancelled context to ensure the remote executions are
+		// canceled
+		_, err = bbClient.CancelExecutions(context.WithoutCancel(ctx), &inpb.CancelExecutionsRequest{
```

Comment:
is this code needed now that it's done in hostedrunner.go?

Reply:
Yeah - the hostedrunner:Run call returns after the invocation has been created. We need this block if you want to cancel as the invocation is running